Description: Fix desktop file: run in window, add keywords
Author: Fabian Greffrath <fabian+debian@greffrath.com>

--- a/misc/rott.desktop
+++ b/misc/rott.desktop
@@ -4,5 +4,6 @@ Name=Rise of the Triad
 Comment=A high quality, fast scrolling first-person perspective 3D action game
 Icon=rott
 TryExec=rott
-Exec=rott -fullscreen -resolution 800x600
+Exec=rott
 Categories=Game;ActionGame;
+Keywords=first;person;shooter;fps;3d;rott;
